astropy/modeling/utils.py

```diff
@@ -744,3 +744,23 @@ def _to_orig_unit(value, raw_unit=None, orig_unit=None):
         return (value * raw_unit).to(orig_unit)
     else:
         return np.rad2deg(value)
+
+
+class _ConstraintsDict(UserDict):
```

Comment:
Why not inherit from dict?

Reply:
I thought it's cleaner to inherit from UserDict since it takes care of some things under the hood. For example, if inheriting from `dict` the `update` method does not work and need to be reimplemented. There might be others but I thought that was sufficient. Do you see a disadvantage of using `UserDict`?

Reply:
OK, I was unaware of that difference (or forgot :-). I thought it was due to the fact that old versions couldn't inherit from dict (but that was probably a long time ago)